lib/block_view_test.go

```diff
@@ -14690,3 +15303,1854 @@ func TestUpdateProfileChangeBack(t *testing.T) {
+			[]byte{},                               /*postHashToModify*/
+			[]byte{},                               /*parentStakeID*/
+			&BitCloutBodySchema{Body: "m0 post 1"}, /*body*/
+			[]byte{},
+			1502947011*1e9, /*tstampNanos*/
+			false /*isHidden*/)
+	}
+	post1Hash := testMeta.txns[len(testMeta.txns)-1].Hash()
+
+	// Error case: can't make an NFT without a profile.
+	{
+		_, _, _, err := _createNFT(
+			t, chain, db, params, 10,
+			m0Pub,
+			m0Priv,
+			post1Hash,
+			100,   /*NumCopies*/
+			false, /*HasUnlockable*/
+			true,  /*IsForSale*/
+			0,     /*MinBidAmountNanos*/
+			0,     /*nftFee*/
+			0,     /*nftRoyaltyToCreatorBasisPoints*/
+			0,     /*nftRoyaltyToCoinBasisPoints*/
+		)
+
+		require.Error(err)
+		require.Contains(err.Error(), RuleErrorCantCreateNFTWithoutProfileEntry)
+	}
+
+	// Create a profile so me can make an NFT.
+	{
+		_updateProfileWithTestMeta(
+			testMeta,
+			10,            /*feeRateNanosPerKB*/
+			m0Pub,         /*updaterPkBase58Check*/
+			m0Priv,        /*updaterPrivBase58Check*/
+			[]byte{},      /*profilePubKey*/
+			"m2",          /*newUsername*/
+			"i am the m2", /*newDescription*/
+			shortPic,      /*newProfilePic*/
+			10*100,        /*newCreatorBasisPoints*/
+			1.25*100*100,  /*newStakeMultipleBasisPoints*/
+			false /*isHidden*/)
+	}
+
+	// Error case: m1 should not be able to turn m0's post into an NFT.
+	{
+		_, _, _, err := _createNFT(
+			t, chain, db, params, 10,
+			m1Pub,
+			m1Priv,
+			post1Hash,
+			100,   /*NumCopies*/
+			false, /*HasUnlockable*/
+			true,  /*IsForSale*/
+			0,     /*MinBidAmountNanos*/
+			0,     /*nftFee*/
+			0,     /*nftRoyaltyToCreatorBasisPoints*/
+			0,     /*nftRoyaltyToCoinBasisPoints*/
+		)
+
+		require.Error(err)
+		require.Contains(err.Error(), RuleErrorCreateNFTMustBeCalledByPoster)
+	}
+
+	// Error case: m0 should not be able to make more than MaxCopiesPerNFT.
+	{
+		_, _, _, err := _createNFT(
+			t, chain, db, params, 10,
+			m0Pub,
+			m0Priv,
+			post1Hash,
+			params.MaxCopiesPerNFT+1, /*NumCopies*/
+			false,                    /*HasUnlockable*/
+			true,                     /*IsForSale*/
+			0,                        /*MinBidAmountNanos*/
+			0,                        /*nftFee*/
+			0,                        /*nftRoyaltyToCreatorBasisPoints*/
+			0,                        /*nftRoyaltyToCoinBasisPoints*/
+		)
+
+		require.Error(err)
+		require.Contains(err.Error(), RuleErrorTooManyNFTCopies)
+	}
+
+	// Error case: m0 should not be able to make an NFT with zero copies.
+	{
+		_, _, _, err := _createNFT(
+			t, chain, db, params, 10,
+			m0Pub,
+			m0Priv,
+			post1Hash,
+			0,     /*NumCopies*/
+			false, /*HasUnlockable*/
+			true,  /*IsForSale*/
+			0,     /*MinBidAmountNanos*/
+			0,     /*nftFee*/
+			0,     /*nftRoyaltyToCreatorBasisPoints*/
+			0,     /*nftRoyaltyToCoinBasisPoints*/
+		)
+
+		require.Error(err)
+		require.Contains(err.Error(), RuleErrorNFTMustHaveNonZeroCopies)
+	}
+
```

Comment:
let's add tests for the following error cases:

1. MinBidAmount is less than 0
2. MinBidAmount is greater than Max value
3. CreatorBasisPoints is less than 0
4. CreatorBasisPoints is greater than max value
5. CoinBasisPoints is less than 0
6. CoinBasisPoints is greater than max value

Reply:
MinBidAmount, CreatorBasisPoints, and CoinBasisPoints are all uint64s in the CreateNFTMetadata.  So I actually cannot set these to be less than 0.

Reply:
Added some more tests for the creator / coin basis points in "TestNFTMoreErrorCases"